product_multi_company/models/product.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2015-2016 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from odoo import api, fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = ["multi.company.abstract", "product.template"]
+    _name = "product.template"
+    _description = "Product Template (Multi-Company)"
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    company_ids = fields.Many2many(
+        string="Companies",
+        comodel_name="res.company",
+        relation="product_variant_companies_rel",
+        compute="_compute_company_ids",
+        store=True,
```

Comment:
**question**: do you think a `compute_sudo=True` is warranted here?

Reply:
From what I understand of the base module, `company_ids` contains all the companies (including the ones the current user cannot see) and it's `company_id` that needs to be computed with the current user.

In general, if it needs to be computed with the current user it shouldn't be stored, and the other way around as well, if it's stored, the result shouldn't change depending on the user that last triggered the computation.

Reply:
(Could this just be a related field? I don't know if it would work that way)

Reply:
It is related by default because of inheritance. Even if you make that relation explicit it doesnt work. As well as with store=False.

Reply:
But by default it's `related` with `store=False`, which doesn't work for direct DB queries. What if we made it `related` with `store=True`?.

Reply:
But why do you mean by "direct queries"? Odoo handles this in ORM space, but not for SQL.

Reply:
Mmh, maybe I misspoke. But I think that somehow the root cause of the issue is that `company_ids` is not a stored field for `product.variant`. Either way, if this solution already works I'm okay with it, I was only wondering if it could be even simpler.

Reply:
It still sounds weird to me that the field `company_ids` would depend on `uid` somehow, and that it doesn't need `compute_sudo`, though. If it works, it works, of course.

Reply:
It shouldn't depend on the current user for sure. The problem is that I'm giving you tips blindly, as I don't understand the reported issue and it doesn't seem related to this.

Reply:
Is there something I can improve in the issue itself to make it clearer? Is there anything in particular that you think is underexplained?

Reply:
The problem is that I can't reproduce the issue in this runboat:

http://oca-multi-company-14-0-d248a27cd7a5.runboat.odoo-community.org/web#id=40&action=145&model=product.product&view_type=form&cids=&menu_id=250

![image](https://github.com/OCA/multi-company/assets/7165771/a115c47f-6b21-4983-9b38-5b7a64e72cb3)

and the error message about company employee rule indicates an issue with other module, not this one.